packages/opentelemetry-plugin-http/src/utils.ts

```diff
@@ -30,20 +31,35 @@ import * as url from 'url';
  */
 export class Utils {
   /**
-   * return an absolute url
+   * Get an absolute url
    */
-  static getUrlFromIncomingRequest(
-    requestUrl: url.UrlWithStringQuery | null,
-    headers: IncomingHttpHeaders
+  static getAbsoluteUrl(
+    requestUrl: ParsedRequestOptions | null,
+    headers: IncomingHttpHeaders | OutgoingHttpHeaders,
+    defaultProtocol = 'http'
   ): string {
     if (!requestUrl) {
-      return `http://${headers.host || 'localhost'}/`;
+      return `${defaultProtocol}://${headers.host || 'localhost'}/`;
     }
 
-    return requestUrl.href && requestUrl.href.startsWith('http')
-      ? `${requestUrl.protocol}//${requestUrl.hostname}${requestUrl.path}`
-      : `${requestUrl.protocol || 'http:'}//${headers.host ||
-          'localhost'}${requestUrl.path || '/'}`;
+    const href = (requestUrl as url.UrlWithParsedQuery).href;
+    if (href && href.startsWith('http')) {
+      return `${requestUrl.protocol}//${requestUrl.hostname}${requestUrl.path}`;
+    }
+
+    const protocol = requestUrl.protocol || `${defaultProtocol}:`;
+    const port = (requestUrl.port || '').toString();
```

Comment:
AFAIK `requestUrl.port` is already type of `string`, which means no need to to `.toString()`

Reply:
With `ClientRequestArgs`, port can be number or string.